src/datasources/queues/queue-consumer.service.ts

```diff
@@ -0,0 +1,41 @@
+import { IConfigurationService } from '@/config/configuration.service.interface';
+import { QueueConsumer } from '@/datasources/queues/queue-consumer.module';
+import { IQueueConsumerService } from '@/datasources/queues/queue-consumer.service.interface';
+import { ILoggingService, LoggingService } from '@/logging/logging.interface';
+import { Inject } from '@nestjs/common';
+import { ConsumeMessage } from 'amqplib';
+
+export class QueueConsumerService implements IQueueConsumerService {
+  private readonly isEventsQueueEnabled: boolean;
+
+  constructor(
+    @Inject('QueueConsumer') private readonly consumer: QueueConsumer,
+    @Inject(LoggingService) private readonly loggingService: ILoggingService,
+    @Inject(IConfigurationService)
+    private readonly configurationService: IConfigurationService,
+  ) {
+    this.isEventsQueueEnabled = this.configurationService.getOrThrow<boolean>(
+      'features.eventsQueue',
+    );
+  }
+
+  isReady(): boolean {
+    if (this.isEventsQueueEnabled) {
+      return this.consumer.connection.isConnected();
+    }
+    return true;
+  }
+
+  async subscribe(
+    queueName: string,
+    fn: (msg: ConsumeMessage) => Promise<void>,
+  ): Promise<void> {
+    if (this.isEventsQueueEnabled) {
+      await this.consumer.channel.consume(queueName, async (msg) => {
+        await fn(msg);
+        await this.consumer.channel.ack(msg);
+      });
+      this.loggingService.info(`Subscribed to queue: ${queueName}`);
+    }
+  }
```

Comment:
Another option besides checking `this.isEventsQueueEnabled` is each function would be injecting a _mock_ `QueueConsumerService` in `QueueConsumerModule`, depending on whether the FF is active or not. I opted for not taking that approach as it would need to make `QueueConsumerModule` dynamic, and given that other parts of the application also depend on it (`HealthRepositoryModule`, `CacheHooksModule`), it would make the overall setup more convoluted in my opinion, also considering this FF would be transitory.